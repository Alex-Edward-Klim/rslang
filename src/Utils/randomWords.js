const random = (maxNum, minNum = 0) => minNum + Math.floor(Math.random() * (maxNum - minNum + 1));

const randomWords = (arr, volume = arr.length) => {
    const vol = volume > arr.length ? arr.length : volume;
    const data = arr.concat();
    const dataRandom = [];
    for (let i = 0; i < vol; i++) {
        dataRandom.push(data.splice(random(data.length - 1), 1)[0]);
    };
    return dataRandom;
}

export default randomWords;