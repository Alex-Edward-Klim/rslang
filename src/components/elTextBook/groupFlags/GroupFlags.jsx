import React from "react";
import "./groupFlags.scss";

function GroupFlags({number, current, handleFlagClick}) {
  let classArea = `textbook-flag__area color-${number}`;
  let classNumber = 'textbook-flag__number'
  if (current === number) {
    classArea += ' current'
    classNumber += ' current'
  }
  return (
    <div className="textbook-flag" onClick={() => handleFlagClick(number)}>
      <div className={classArea}></div>
      <div className={classNumber}>{number + 1}</div>
    </div>
  )
}

export default GroupFlags