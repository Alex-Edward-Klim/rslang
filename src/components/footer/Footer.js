import React from "react";
import styles from './footer.module.css';
import logoSrc from "../../images/icons/rs_school_js.svg";

function Footer() {
    return (
    <footer className={styles.footer}>
        <div className={styles.footerContainer}>
        <a className={styles.footerLogo} href="https://rs.school/js/">
            <img className={styles.footerLogoImg} alt="rs school" src={logoSrc} />
        </a>
        <p className={styles.footerAuthors}>
            <a href="https://github.com/Alex-Edward-Klim">
            <span>Алексей Климовской</span>
            </a>
            ,
            <a href="https://github.com/korytsa">
            <span> Анна Корытько</span>
            </a>
            ,
            <a href="https://github.com/YaroslavTrefilov">
            <span> Ярослав Трефилов</span>
            </a>
            ,
            <a href="https://github.com/NachinkaShaurmi">
            <span> Юрий Олейник</span>
            </a>, 
            <a href="https://github.com/AV-63-dev">
            <span> Анатолий Манин</span>
            </a>
        </p>
        <a className={styles.footerRss} href="https://rs.school/js/">
            <p>RS School</p>
            <p>2021</p>
        </a>
        </div>
    </footer>
    );
}

export default Footer;