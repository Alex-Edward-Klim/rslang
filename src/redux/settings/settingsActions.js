import { SET_SETTINGS } from "./settingsTypes";

// Action Creators
export const setSettings = (data) => ({
  type: SET_SETTINGS,
  payload: data,
});
