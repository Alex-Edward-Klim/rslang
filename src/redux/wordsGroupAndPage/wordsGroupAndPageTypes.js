// Action Types
export const SET_WORDS_GROUP_AND_PAGE = "SET_WORDS_GROUP_AND_PAGE";
