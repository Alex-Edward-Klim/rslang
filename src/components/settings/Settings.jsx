import React from "react";
import { useSelector, useDispatch } from "react-redux";
import Checkbox from "@material-ui/core/Checkbox";
import { getSettingsFromState } from "../../redux/selectors";
import { setSettings } from "../../redux/settings/settingsActions";
import s from './settings.module.scss'


function Settings() {
  const { translate, changeWordStatus } = useSelector(getSettingsFromState);
  const dispatch = useDispatch();

  const handleSetting = (name, value) => {
    dispatch(setSettings({ translate, changeWordStatus, [name]: value }))
  };

  return (
    <>
      <div className={s.settings}>
        <h2 className={s.title}>Настройки</h2>
        <form action="" className={s.form}>
          <label className={s.label}>
            <Checkbox
              color="primary"
              name="translate"
              onChange={({ target }) =>
                handleSetting(target.name, target.checked)
              }
              checked={translate}
            />
            Показывать перевод
          </label>

          <label className={s.label}>
            <Checkbox
              color="primary"
              name="changeWordStatus"
              onChange={({ target }) =>
                handleSetting(target.name, target.checked)
              }
              checked={changeWordStatus}
            />
            Показывать кнопки добавления в разделы словаря
          </label>
        </form>
      </div>
    </>
  );
}

export default Settings;
