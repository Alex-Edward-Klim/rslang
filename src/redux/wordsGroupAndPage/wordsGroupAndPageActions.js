import { SET_WORDS_GROUP_AND_PAGE } from "./wordsGroupAndPageTypes";

// Action Creators
export const setWordsGroupAndPage = (obj) => {

  localStorage.setItem(
    "RSLangWordsGroupAndPage78fe8a83ef752bd23c98c262b7264947",
    JSON.stringify(obj)
  );
  
  return {
    type: SET_WORDS_GROUP_AND_PAGE,
    payload: obj,
  };
};
