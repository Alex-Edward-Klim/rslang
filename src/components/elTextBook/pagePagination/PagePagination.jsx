import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { getWordsGroupAndPageFromState } from "../../../redux/selectors";
import { setWordsGroupAndPage } from "../../../redux/wordsGroupAndPage/wordsGroupAndPageActions";
import "./pagePagination.scss";
import arrowL from "../../../img/arrowL.png";
import arrowR from "../../../img/arrowR.png";

function PagePagination({maxPage}) {
  const { group, page } = useSelector(getWordsGroupAndPageFromState);
  const dispatch = useDispatch();

  let buttonLeftClass = "page-pagination__block left";
  if (page === 0) buttonLeftClass += " disable";
  let buttonRightClass = "page-pagination__block right";
  if (page === maxPage) buttonRightClass += " disable";

  return (
    <div className="page-pagination">
      <button
        className={buttonLeftClass}
        disabled={page === 0}
        onClick={() =>
          dispatch(setWordsGroupAndPage({ group, page: page - 1 }))
        }
      >
        <img src={arrowL} alt="" width="30" />
      </button>
      <div className="page-pagination__block  number">{page + 1}</div>
      <button
        className={buttonRightClass}
        disabled={page === maxPage}
        onClick={() =>
          dispatch(setWordsGroupAndPage({ group, page: page + 1 }))
        }
      >
        <img src={arrowR} alt="" width="30" />
      </button>
    </div>
  );
}

export default PagePagination;
