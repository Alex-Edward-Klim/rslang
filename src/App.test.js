import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import PagePagination from "./components/elTextBook/pagePagination/PagePagination";
import Footer from "./components/footer/Footer";
import GroupFlags from "./components/elTextBook/groupFlags/GroupFlags";
import randomWords from "./Utils/randomWords";
import { SET_USER_DATA } from "./redux/user/userTypes";
import { SET_SETTINGS } from "./redux/settings/settingsTypes";
import { SET_WORDS_GROUP_AND_PAGE } from "./redux/wordsGroupAndPage/wordsGroupAndPageTypes";
import { setUserData } from "./redux/user/userActions";
import { setSettings } from "./redux/settings/settingsActions";
import { setWordsGroupAndPage } from "./redux/wordsGroupAndPage/wordsGroupAndPageActions";

describe("Snapshot component", () => {
  it("Footer", () => {
    const rendered = render(<Footer />);
    expect(rendered).toMatchSnapshot();
  });

  it("GroupFlags", () => {
    const rendered = render(
      <GroupFlags number={1} current={0} key={1} handleFlagClick={() => {}} />
    );
    expect(rendered).toMatchSnapshot();
  });
});

describe("Function", () => {
  it("randomWords1", () => {
    expect(randomWords(["test1", "test2", "test3"])).toHaveLength(3);
  });
  it("randomWords2", () => {
    expect(randomWords(["test1", "test2", "test3"])).toContain("test1");
  });
  it("randomWords3", () => {
    expect(randomWords(["test1", "test2", "test3"])).toContain("test2");
  });
  it("randomWords4", () => {
    expect(randomWords(["test1", "test2", "test3"])).toContain("test3");
  });
});

describe("Redux types", () => {
  it("SET_USER_DATA", () => {
    expect(SET_USER_DATA).toBe("SET_USER_DATA");
  });
  it("SET_WORDS_GROUP_AND_PAGE", () => {
    expect(SET_WORDS_GROUP_AND_PAGE).toBe("SET_WORDS_GROUP_AND_PAGE");
  });
  it("SET_SETTINGS", () => {
    expect(SET_SETTINGS).toBe("SET_SETTINGS");
  });
});

describe("Redux actions", () => {
  it("User action", () => {
    expect(setUserData("test0")).toEqual({
      type: SET_USER_DATA,
      payload: "test0",
    });
  });
  it("Setting action", () => {
    expect(setSettings("test1")).toMatchObject({
      payload: "test1",
    });
  });
  it("WordsGroupAndPage action", () => {
    expect(setWordsGroupAndPage("test2")).toMatchObject({
      payload: "test2",
    });
  });
});

const mockStore = configureStore([]);

describe("PagePagination React-Redux Component", () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      wordsGroupAndPage: {
        group: 0,
        page: 0,
      },
    });

    component = renderer.create(
      <Provider store={store}>
        <PagePagination maxPage={5} />
      </Provider>
    );
  });

  it("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
