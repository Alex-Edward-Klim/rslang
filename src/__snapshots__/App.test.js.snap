// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagePagination React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="page-pagination"
>
  <button
    className="page-pagination__block left disable"
    disabled={true}
    onClick={[Function]}
  >
    <img
      alt=""
      src="arrowL.png"
      width="30"
    />
  </button>
  <div
    className="page-pagination__block  number"
  >
    1
  </div>
  <button
    className="page-pagination__block right"
    disabled={false}
    onClick={[Function]}
  >
    <img
      alt=""
      src="arrowR.png"
      width="30"
    />
  </button>
</div>
`;

exports[`Snapshot component Footer 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="footer"
      >
        <div
          class="footerContainer"
        >
          <a
            class="footerLogo"
            href="https://rs.school/js/"
          >
            <img
              alt="rs school"
              class="footerLogoImg"
              src="rs_school_js.svg"
            />
          </a>
          <p
            class="footerAuthors"
          >
            <a
              href="https://github.com/Alex-Edward-Klim"
            >
              <span>
                Алексей Климовской
              </span>
            </a>
            ,
            <a
              href="https://github.com/korytsa"
            >
              <span>
                 Анна Корытько
              </span>
            </a>
            ,
            <a
              href="https://github.com/YaroslavTrefilov"
            >
              <span>
                 Ярослав Трефилов
              </span>
            </a>
            ,
            <a
              href="https://github.com/NachinkaShaurmi"
            >
              <span>
                 Юрий Олейник
              </span>
            </a>
            ,
            <a
              href="https://github.com/AV-63-dev"
            >
              <span>
                 Анатолий Манин
              </span>
            </a>
          </p>
          <a
            class="footerRss"
            href="https://rs.school/js/"
          >
            <p>
              RS School
            </p>
            <p>
              2021
            </p>
          </a>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="footer"
    >
      <div
        class="footerContainer"
      >
        <a
          class="footerLogo"
          href="https://rs.school/js/"
        >
          <img
            alt="rs school"
            class="footerLogoImg"
            src="rs_school_js.svg"
          />
        </a>
        <p
          class="footerAuthors"
        >
          <a
            href="https://github.com/Alex-Edward-Klim"
          >
            <span>
              Алексей Климовской
            </span>
          </a>
          ,
          <a
            href="https://github.com/korytsa"
          >
            <span>
               Анна Корытько
            </span>
          </a>
          ,
          <a
            href="https://github.com/YaroslavTrefilov"
          >
            <span>
               Ярослав Трефилов
            </span>
          </a>
          ,
          <a
            href="https://github.com/NachinkaShaurmi"
          >
            <span>
               Юрий Олейник
            </span>
          </a>
          ,
          <a
            href="https://github.com/AV-63-dev"
          >
            <span>
               Анатолий Манин
            </span>
          </a>
        </p>
        <a
          class="footerRss"
          href="https://rs.school/js/"
        >
          <p>
            RS School
          </p>
          <p>
            2021
          </p>
        </a>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot component GroupFlags 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="textbook-flag"
      >
        <div
          class="textbook-flag__area color-1"
        />
        <div
          class="textbook-flag__number"
        >
          2
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="textbook-flag"
    >
      <div
        class="textbook-flag__area color-1"
      />
      <div
        class="textbook-flag__number"
      >
        2
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
